virtual_rainforest/core/config.py

```diff
@@ -0,0 +1,362 @@
+                f"should either be renamed or deleted!",
+                OSError,
+            )
+
+    return None
+
+
+def collect_files(cfg_paths: list[str]) -> list[Path]:
+    """Collect all toml files from a user specified list of files and directories.
+
+    Args:
+        cfg_paths: A path or a set of paths that point to either configuration files, or
+            folders containing configuration files
+    """
+
+    # Preallocate file list
+    files = []
+    not_found = []  # Stores all invalid paths
+    empty_fold = []  # Stores all empty toml folders
+
+    for path in cfg_paths:
+        p = Path(path)
+        # First check that each file is valid
+        if p.exists():
+            # Check if each path is to a file or a directory
+            if p.is_dir():
+                toml_files = list([f for f in p.glob("*.toml")])
+                if len(toml_files) != 0:
+                    files.extend(toml_files)
+                else:
+                    empty_fold.append(path)
+            else:
+                files.append(Path(path))
+
+        else:
+            # Add missing path to list of missing paths
+            not_found.append(path)
+
+    # Check for items that are not found
+    if len(not_found) != 0:
+        log_and_raise(
+            f"The following (user provided) config paths do not exist:\n{not_found}",
+            OSError,
+        )
+    # And for empty folders
+    elif len(empty_fold) != 0:
+        log_and_raise(
+            f"The following (user provided) config folders do not contain any toml "
+            f"files:\n{empty_fold}",
+            OSError,
+        )
+    # Finally check that no files are pointed to twice
+    elif len(files) != len(set(files)):
+        log_and_raise(
+            f"A total of {len(files) - len(set(files))} config files are specified more"
+            f" than once (possibly indirectly)",
+            RuntimeError,
+        )
+
+    return files
+
+
+def load_in_config_files(files: list[Path]) -> dict:
+    """Load in a set of toml files checking that no tags are repeated.
+
+    This function also ensure that no tags are repeated across different toml files.
+
+    Args:
+        files: List of files to be read in and checked for overlapping tags
+    """
+
+    # Preallocate container for file names and corresponding dictionaries
+    file_data: list[tuple[Path, dict]] = []
+    conflicts = []
+
+    # Load all toml files that we want to read from
+    for file in files:
+        # If not then read in the file data
+        with open(file, "rb") as f:
+            try:
+                toml_dict = tomllib.load(f)
+                # Check for repeated entries across previous nested dictionaries
+                for item in file_data:
+                    repeats = check_dict_leaves(item[1], toml_dict, [])
+                    for elem in repeats:
+                        conflicts.append((elem, file, item[0]))
+
+                file_data.append((file, toml_dict))
```

Comment:
if changed to a dict:
```suggestion
                file_data[file] = toml_dict
```